coap/adapter.go

```diff
@@ -99,6 +104,11 @@ func (svc *adapterService) Subscribe(ctx context.Context, key, domainID, chanID,
 		return svcerr.ErrAuthentication
 	}
 
+	// Health topics do not subscribe to the message broker.
+	if topicType == messaging.HealthType {
+		return nil
+	}
```

Comment:
Let's only have health check for publishing, i.e. keep it as simple as possible.